### ISSUE TITLE
Maximum infiltration volume (Modified Horton method)

### ISSUE THREAD

**Issue by MikeGregory-Canada** [USEPA/Stormwater-Management-Model#59]
The Maximum Infiltration Volume feature works fine with the original Horton infiltration model (it correctly shuts off infiltration at 1-5 inches for the various subcatchments in the attached file). But it gives incorrect results when you switch to Modified Horton.

[Horton_MaxVolumeTest.zip](https://github.com/USEPA/Stormwater-Management-Model/files/5143913/Horton_MaxVolumeTest.zip)

**Comment by dickinsonre** [comment]
It seems as if Fmax is used for both Horton and Modified Horton in infil.c - is it being exported for the Modified Horton option?

**Comment by MikeGregory-Canada** [comment]
The only difference is the Infiltration method in the [OPTIONS] section of the input file. 
![InputFileDiff](https://user-images.githubusercontent.com/70409362/91613790-3be7f680-e94e-11ea-8574-21b0952b7416.png)

**Comment by LRossman** [comment]
This bug has now been fixed in PR #56 . Thanks @MikeGregory-Canada .

![image](https://user-images.githubusercontent.com/12769926/91639502-05a98600-e9e5-11ea-9a2c-e43895c39335.png)

**Comment by dickinsonre** [comment]
Yes, thanks @MikeGregory-Canada  for finding this bug from Canada.

**Comment by joelnc** [comment]
Entirely possible I'm missing something here, but is there any chance this issue has somehow reverted?  

In the 5.2.4 GUI, using a simple example model with six subcatchments incremented up on Max Vol. from 0 to 5", via traditional Horton I get:

![image](https://github.com/USEPA/Stormwater-Management-Model/assets/20247905/65e165c9-9f19-4d23-91ca-75df1442868f)

But when I toggle the infiltration model to Modified Horton I get:

![image](https://github.com/USEPA/Stormwater-Management-Model/assets/20247905/445aa352-2d9c-4105-b518-6f90ee562821)

Model details:

```
[SUBCATCHMENTS]
;;Name           Rain Gage        Outlet           Area     %Imperv  Width    %Slope   CurbLen  SnowPack        
;;-------------- ---------------- ---------------- -------- -------- -------- -------- -------- ----------------
MaxVol_0         RG1              9                10       10       500      0.01     0                        
MaxVol_1         RG1              10               10       10       500      0.01     0                        
MaxVol_2         RG1              13               10       10       500      0.01     0                        
MaxVol_3         RG1              22               10       10       500      0.01     0                        
MaxVol_4         RG1              15               10       10       500      0.01     0                        
MaxVol_5         RG1              23               10       10       500      0.01     0                        

[SUBAREAS]
;;Subcatchment   N-Imperv   N-Perv     S-Imperv   S-Perv     PctZero    RouteTo    PctRouted 
;;-------------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
MaxVol_0         0.001      0.10       0.05       0.20       25         OUTLET    
MaxVol_1         0.001      0.10       0.05       0.20       25         OUTLET    
MaxVol_2         0.001      0.10       0.05       0.20       25         OUTLET    
MaxVol_3         0.001      0.10       0.05       0.20       25         OUTLET    
MaxVol_4         0.001      0.10       0.05       0.20       25         OUTLET    
MaxVol_5         0.001      0.10       0.05       0.20       25         OUTLET    

[INFILTRATION]
;;Subcatchment   Param1     Param2     Param3     Param4     Param5    
;;-------------- ---------- ---------- ---------- ---------- ----------
MaxVol_0         1.50       0.3        4.14       0.50       0         
MaxVol_1         1.50       0.3        4.14       0.50       1         
MaxVol_2         1.50       0.3        4.14       0.50       2         
MaxVol_3         1.50       0.3        4.14       0.50       3         
MaxVol_4         1.50       0.3        4.14       0.50       4         
MaxVol_5         1.50       0.3        4.14       0.50       5         

```

**Comment by joelnc** [comment]
@dickinsonre Thanks Bob.  Can I read your thumbs up as "issue confirmed"?  I'm always hesitant to rule out the possibility of simple user error, especially when the user is @joelnc.  If this functionality is truly broken in 5.2.4, then I can just move ahead without it for now.

**Comment by cbuahin** [comment]
@joelnc, it appears you are right. The fix was not propagated to 5.2.4 somehow.  I am trying to figure out where this divergence came about. I will reopen this ticket in the meantime.

**Comment by joelnc** [comment]
Thanks Caleb.

**Comment by LRossman** [comment]
For some unknown reason the Modified Horton bug fix in PR #56 never actually got added to release 5.1.015 and all subsequent releases. The attached zip file contains the updates to infil.h and infil.c that fix the bug (tested on example1.inp by setting the max. infiltration volume to 2" for all subcatchments).

[infil.zip](https://github.com/USEPA/Stormwater-Management-Model/files/13641453/infil.zip)

**Comment by joelnc** [comment]
Thank you all.  Just completed some runs with the bug_fix branch, and the functionality is now working as expected.
